File: app/src/test/java/com/mgaetan89/showsrage/fragment/AddShowFragment_OnQueryTextChangeTest.java
package com.mgaetan89.showsrage.fragment;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;

import java.util.Collection;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(Parameterized.class)
public class AddShowFragment_OnQueryTextChangeTest {
	@Parameterized.Parameter(0)
	public String query;

	@Parameterized.Parameter(1)
	public boolean valid;

	private AddShowFragment fragment;

	@Before
	public void before() {
		this.fragment = new AddShowFragment();
	}

	@Test
	public void onQueryTextChange() {
		assertThat(this.fragment.onQueryTextChange(this.query)).isFalse();
	}

	@After
	public void after() {
		this.fragment = null;
	}

	@Parameterized.Parameters
	public static Collection<Object[]> data() {
		return AddShowFragment_IsQueryValidTest.data();
	}
}

File: app/src/test/kotlin/com/mgaetan89/showsrage/fragment/AddShowFragment_OnQueryTextChangeTest.kt
package com.mgaetan89.showsrage.fragment

import org.assertj.core.api.Assertions.assertThat
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized

@RunWith(Parameterized::class)
class AddShowFragment_OnQueryTextChangeTest(val query: String?, val valid: Boolean) {
	private lateinit var fragment: AddShowFragment

	@Before
	fun before() {
		this.fragment = AddShowFragment()
	}

	@Test
	fun onQueryTextChange() {
		assertThat(this.fragment.onQueryTextChange(this.query)).isFalse()
	}

	companion object {
		@JvmStatic
		@Parameterized.Parameters
		fun data() = AddShowFragment_IsQueryValidTest.data()
	}
}
